### ISSUE TITLE
We currently include stop codons in protein sequences

### ISSUE THREAD

**Issue by theosanderson** [loculus-project/loculus#2841]
At least in production, we currently include stop codons (`*`) at the end of our protein sequences. I think to me this feels quite unexpected. NCBI doesn't seem to do it: https://www.ncbi.nlm.nih.gov/protein/YP_011036812 .

We should think through what the ideal behaviour on this is.

**Comment by chaoran-chen** [comment]
Interesting to see that NCBI doesn't! Nextclade always includes the stop codon if I'm not mistaken, so I think that I have only ever worked with protein sequences that end with it. Do you see pros/cons for including/not including it?

**Comment by theosanderson** [comment]
It would be cool to ask the Nextclade/Nextstrain folks as I guess they will have views on the pros of including it.

Basically, I find it confusing that it is included. I think of these as amino acid sequences, and `stop` isn't an amino acid. UniProt is the main protein sequence database and also doesn't include them https://www.uniprot.org/uniprotkb/Q6SJ61/entry#sequences (and same for NCBI as discussed). It feels redundant given that it's always this ending. It just confused me when I was adding a new prototype organism and forgot to include these in the yaml, and the same happened with one of our collaborators who was testing Loculus for their organism. ChatGPT (fwiw) says that not including them is more common, and mentions the PDB too which is another point of comparison https://chatgpt.com/share/66eb36d9-782c-8005-9126-a24d3c105a6b .

**Comment by chaoran-chen** [comment]
ok, interesting! I have no preference regarding this but a general question: if we decide to (or not to) change this for Pathoplexus / the Nextclade-based pipeline, do you think that we should regulate this on the Loculus-level, i.e., enforce that an AA sequence must or must not end with a stop codon?

**Comment by corneliusroemer** [comment]
One reason to include them is that this way one can detect when the stop is no longer a stop. I don't know for sure why we do what we do in Nextclade, but this may be a reason. @ivan-aksamentov @rneher will know better.

**Comment by theosanderson** [comment]
Yes, to be clear I'm definitely not proposing any change to Nextclade.

Thanks, that's useful. Yes, I hadn't fully appreciated that the CDS is annotated up to the stop codon - so with the NCBI approach your nucleotide sequence isn't exactly 3x as long as your amino acid sequence, which seems bad too.

**Comment by rneher** [comment]
I like having the stop. but it is a personal preference, I think. For viruses, this results in a useful distinction:

 - peptides that are the result of a polypeptide cleaved by a protease won't have a stop
 - those that come straight from the ribosome do. 

Biopython includes the terminal `*`, unless you specify `to_stop` or `cds=True`
```
Seq.translate(
    sequence,
    table='Standard',
    stop_symbol='*',
    to_stop=False,
    cds=False,
    gap=None,
)
Docstring:
Translate a nucleotide sequence into amino acids.

If given a string, returns a new string object. Given a Seq or
MutableSeq, returns a Seq object.

Arguments:
 - table - Which codon table to use?  This can be either a name
   (string), an NCBI identifier (integer), or a CodonTable object
   (useful for non-standard genetic codes).  Defaults to the "Standard"
   table.
 - stop_symbol - Single character string, what to use for any
   terminators, defaults to the asterisk, "*".
 - to_stop - Boolean, defaults to False meaning do a full
   translation continuing on past any stop codons
   (translated as the specified stop_symbol).  If
   True, translation is terminated at the first in
   frame stop codon (and the stop_symbol is not
   appended to the returned protein sequence).
 - cds - Boolean, indicates this is a complete CDS.  If True, this
   checks the sequence starts with a valid alternative start
   codon (which will be translated as methionine, M), that the
   sequence length is a multiple of three, and that there is a
   single in frame stop codon at the end (this will be excluded
   from the protein sequence, regardless of the to_stop option).
   If these tests fail, an exception is raised.
 - gap - Single character string to denote symbol used for gaps.
   Defaults to None.
```
